bundles/org.openhab.binding.opensprinkler/README.md

```diff
@@ -61,9 +61,11 @@ When using the `nextDuration` channel, it is advised to setup persistence (e.g.
 
 The following is supported by the `device` thing, but only when connected using the http interface.
 
-| Channel Type ID | Item Type |    | Description                                                           |
-|-----------------|-----------|----|-----------------------------------------------------------------------|
-| rainsensor      | Switch    | RO | This channel indicates whether rain is detected by the device or not. |
+| Channel Type ID | Item Type              |    | Description                                                               |
+|-----------------|------------------------|----|---------------------------------------------------------------------------|
+| rainsensor      | Switch                 | RO | This channel indicates whether rain is detected by the device or not.     |
+| currentDraw     | Number:ElectricCurrent | RO | Shows the current draw of the device. If the device does not have sensors |
```

Comment:
Is it possible to detect if the device has the capability?

Reply:
Most likely, if the `curr` key in the API does not exist. Why, would it be possible to conditionally add the channel?

Reply:
It is possible to dynamically add channels. I know it is done in the bluetooth binding ConnectedBluetoothHandler but I'm not sure it is the best example, it is just the only example I can think of ATM.
Dynamically adding channels is essentially modifying the Thing configuration in runtime.

Reply:
I think the Daikin binding has this now too

Reply:
Exactly. Once you know you need to add a channel, you do something like

```
ThingBuilder thingBuilder = editThing();
ChannelBuilder channelBuilder = ChannelBuilder.create(
    new ChannelUID(thing.getUID(), "currentDraw"), "Number:ElectricCurrent")
    .withType(<insert type here>).withLabel(<add chabnnel label here>);
thingBuilder.withChannel(channelBuilder.build);
updateThing(thingBuilder.build());
```

or to remove a channel

```
ThingBuilder thingBuilder = editThing();
if (thing.getChannel(channelId) != null) {
    thingBuilder.withoutChannel(new ChannelUID(thing.getUID(), channelId));
}
updateThing(thingBuilder.build());
```

Reply:
Ok, then let's try to add it conditionally :)